packages/terra-date-picker/src/react-datepicker/index.jsx

```diff
@@ -51,84 +51,297 @@ const WrappedCalendar = onClickOutside(Calendar)
+    /**
+     * A callback function to execute when date is entered.
+     */
     onChangeRaw: PropTypes.func,
+    /**
+     * A callback function to execute when object is in focus.
+     */
     onFocus: PropTypes.func,
+    /**
+     * A callback function to execute when a key is pressed.
+     */
     onKeyDown: PropTypes.func,
+    /**
+     * A callback function to execute when month is selected.
+     */
     onMonthChange: PropTypes.func,
+    /**
+     * Prop to open calendar on a particular date.
+     */
     openToDate: PropTypes.object,
+    /**
+     * A prop to check next month dates.
+     */
     peekNextMonth: PropTypes.bool,
+    /**
+     * Value for placeholder of date picker.
+     */
     placeholderText: PropTypes.string,
+    /**
+     * Prop to prevent date picker from opening on click.
+     */
     preventOpenOnFocus: PropTypes.bool,
+    /**
+     * Prop to make date picker as read only.
+     */
     readOnly: PropTypes.bool,
+    /**
+     * Prop to allow as required field.
+     */
     required: PropTypes.bool,
+    /**
+     * Prop to allow scrollable year dropdown.
+     */
     scrollableYearDropdown: PropTypes.bool,
+    /**
+     * Selected Date Value.
+     */
     selected: PropTypes.object,
+    /**
+     * Prop to select end date on a date picker.
+     */
     selectsEnd: PropTypes.bool,
+    /**
+     * Prop to select start date on a date picker.
+     */
     selectsStart: PropTypes.bool,
+    /**
+     * Prop to show month drop down in calendar.
+     */
     showMonthDropdown: PropTypes.bool,
+    /**
+     * Prop to show week numbers .
+     */
     showWeekNumbers: PropTypes.bool,
+    /**
+     * Prop to show a dropsown to select year in date picker calendar .
+     */
     showYearDropdown: PropTypes.bool,
+    /**
+     * Prop to show a month navigation in date picker .
+     */
     forceShowMonthNavigation: PropTypes.bool,
+    /**
+     * Date selected by user .
+     */
     startDate: PropTypes.object,
+    /**
+     * Prop to open calendar dropdown .
+     */
     startOpen: PropTypes.bool,
+    /**
+     * Prop to specify tabbing order of elements .
+     */
     tabIndex: PropTypes.number,
+    /**
+     * Prop to specify title attribute for date picker .
+     */
     title: PropTypes.string,
+    /**
+     * Name of button to select current date .
+     */
     todayButton: PropTypes.string,
+    /**
+     * Prop to show short names for weekdays on calendar .
+     */
     useWeekdaysShort: PropTypes.bool,
+    /**
+     * Difference between utc and local time.
+     */
     utcOffset: PropTypes.number,
+    /**
+     * Value of the date picked by user .
+     */
     value: PropTypes.string,
+    /**
+     * Label value for weeks on date picker.
+     */
     weekLabel: PropTypes.string,
+    /**
+     * Prop to show a separate portal version for date picker.
+     */
     withPortal: PropTypes.bool,
+    /**
+     * Year Values to show on dropdown +/- the given value.
```

Comment:
This could be changed to https://github.com/cerner/terra-framework/pull/785/files#diff-c35e15214fd200ac057443cf11375217R40.